app/javascript/shared/tubeTransferVolumes.js

```diff
@@ -1,22 +1,24 @@
-const purposeTargetMolarityParameter = function(purposeConfig) {
-  return purposeConfig?.transfer_parameters?.target_molarity_nm
-}
+const purposeTargetMolarityParameter = (purposeConfig) =>
+  purposeConfig?.transfer_parameters?.target_molarity_nm
 
-const purposeTargetVolumeParameter = function(purposeConfig) {
-  return purposeConfig?.transfer_parameters?.target_volume_ul
-}
+const purposeTargetVolumeParameter = (purposeConfig) =>
+  purposeConfig?.transfer_parameters?.target_volume_ul
 
-const purposeMinimumPickParameter = function(purposeConfig) {
-  return purposeConfig?.transfer_parameters?.minimum_pick_ul
-}
+const purposeMinimumPickParameter = (purposeConfig) =>
+  purposeConfig?.transfer_parameters?.minimum_pick_ul
 
 const tubeMostRecentMolarity = function(tube) {
-  // Get the QC results, find those for molarity entries in nM, inverse sort by the created_at timestamp
-  // then take the first item.
-  return tube?.receptacle?.qc_results
-    ?.filter(result => result.key === 'molarity' && result.units === 'nM')
-    .sort((resultA, resultB) => -1 * ('' + resultA.created_at).localeCompare(resultB.created_at))[0]
-    ?.value
+  // Get the QC results,
+  // find those for molarity entries in nM,
+  // inverse sort by the created_at timestamp,
+  // inverse sort by the id,
+  // take the first item and parse the value to a float.
+  const qcResults = tube?.receptacle?.qc_results
+  const molarityEntries =  qcResults?.filter(result => result.key === 'molarity' && result.units === 'nM')
+  const sortedByCreatedAt = molarityEntries?.sort((resultA, resultB) => -1 * ('' + resultA.created_at).localeCompare(resultB.created_at))
+  const sortedByIds = sortedByCreatedAt?.sort((resultA, resultB) => parseInt(resultA.id) > parseInt(resultB.id) ? -1 : 1)
```

Comment:
Was about to point out that the ECMA script standard didn't specify a stable sort, but it looks like this has changed. I guess it may mean that the won't work on some ancient browsers, but given id will show the correct behaviour in almost all cases, I don't think its worth making things more complicated here. (Especially as I'd be incredibly surprised to find an implementation that wasn't stable)